dotnet/src/webdriver/Remote/HttpCommandExecutor.cs

```diff
@@ -385,5 +393,43 @@ private class HttpResponseInfo
             public string Body { get; set; }
             public string ContentType { get; set; }
         }
+
+        /// <summary>
+        /// Internal Diagnostic Handler for HttpCommandExecutor Additional Context
+        /// </summary>
+        internal class DiagnosticsHttpHandler : DelegatingHandler
+        {
+            private static readonly ILogger _logger = Log.GetLogger<DiagnosticsHttpHandler>();
+
+            public DiagnosticsHttpHandler(HttpMessageHandler messageHandler)
+                : base(messageHandler)
+            {
+            }
+
+            /// <summary>
+            /// Sends the specified request and returns the associated response.
+            /// </summary>
+            /// <param name="request">The request to be sent.</param>
+            /// <param name="cancellationToken">A CancellationToken object to allow for cancellation of the request.</param>
+            /// <returns>The http response message content.</returns>
+            protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                if (request.Content != null)
+                {
+                    var requestContent = await request.Content.ReadAsStringAsync().ConfigureAwait(false);
+                    _logger.Trace($">> Body: {requestContent}");
+                }
+
+                var response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```

Comment:
`response.Content` might be `null` here.

Reply:
Can we resolve this one?